sound/soc/sof/intel/hda-dsp.c

```diff
@@ -412,7 +412,7 @@ static int hda_dsp_set_D0_state(struct snd_sof_dev *sdev,
 		 * But it can be kept enabled when the DSP enters D0I3 while the
 		 * system is in S0 for debug.
 		 */
-		if (hda_enable_trace_D0I3_S0 &&
+		if (!hda_enable_trace_D0I3_S0 ||
```

Comment:
The command says 'trace DMA is disabled by default'.
but you enable it by default in the kernel parameter.
One of the two is wrong then?

Reply:
Thanks, the line 411 actually should be "Trace DMA need to be disabled when the DSP enters D0I3 for S0Ix suspend".

Reply:
then the value also needs to be 'false'?

Reply:
you mean the value of hda_enable_trace_D0I3_S0? No, it denotes the value for S0 (not for S0Ix), we want to make DMA trace enabled by default on S0 for KWD (D0I3), when DMA trace is available, just same with other normal S0 scenarios.

Reply:
@ranj063 did we really intend to have traces by default in S0/D0i3?

Reply:
@plbossart no I dont think so. It is a debug feature and should only be enabled if requested.

Reply:
@ranj063 @plbossart I would ask the question in this way:

do we intend to have traces by default in S0/D0i3 when dtrace_is_supported is true (CONFIG_SND_SOC_SOF_DEBUG_ENABLE_FIRMWARE_TRACE=y, dma trace is enabled for normal S0)?

Or actually we should discard this "hda_enable_trace_D0I3_S0" item so no need for checking it (same with hda_enable_trace_D0I3_S0==true by default)?